stagelib/envglobals.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2017 SUSE LLC
+
+import logging
+import os
+import sys
+
+try:
+    CEPH_VERSION = os.environ['CEPH_VERSION']
+    OS_NAME = os.environ['OS_NAME']
+    OS_VERSION = os.environ['OS_VERSION']
+    BASEOS_REG = os.environ['BASEOS_REG']
+    BASEOS_REPO = os.environ['BASEOS_REPO']
+    BASEOS_TAG = os.environ['BASEOS_TAG']
+    ARCH = os.environ['ARCH']
+    IMAGES_TO_BUILD = os.environ['IMAGES_TO_BUILD'].split(' ')
+    STAGING_DIR = os.environ['STAGING_DIR']
+except KeyError as k:
+    unset_var = k.args[0]
+    errtext = """
+Expected environment variable '{}' to be set.
+Required environment variables:
+ - CEPH_VERSION - Ceph named version being built (e.g., luminous, mimic)
+ - ARCH - Architecture of binaries being built (e.g., amd64, arm32, arm64)
+ - OS_NAME - OS name as used by the ceph-container project (e.g., ubuntu, opensuse)
+ - OS_VERSION - OS version as used by the ceph-container project (e.g., 16.04, 42.3 respectively)
+ - BASEOS_REG - Registry for the container base image (e.g., _ (default reg), arm32v7, arm64v8)
+                There is a relation between binaries built (ARCH) and this value
+ - BASEOS_REPO - Repository for the container base image (e.g., ubuntu, opensuse, alpine)
+ - BASEOS_TAG - Tagged version of the BASEOS_REPO container (e.g., 16.04, 42.3, 3.6 respectively)
+ - IMAGES_TO_BUILD - Container images to be built (usually should be 'dockerfile daemon')
+ - STAGING_DIR - Dir into which files will be staged
+                 This dir will be overwritten if it already exists
+
+"""
+    sys.stderr.write(errtext.format(unset_var))
+    sys.exit(1)
+
+logger = logging.getLogger(__name__)
+
+
+def printGlobal(var):
+    varname = [name for name in globals() if globals()[name] is var][0]
```

Comment:
This could be more straightforward `globals()['var_name']`

Reply:
yeah, if anything it seems to me that the "is var" works from luck, as literals are implemented as global shared objects, so there is only one instance of "1", for example.  anyway...can't think of a reason to search by value.

Reply:
I just did a quick test of this, and since strings are immutable in Python, you are right. And it's on my to-fix list.